extensions/pyRevitTools.extension/checks/radar_check.py

```diff
@@ -0,0 +1,624 @@
+    tbdata = [['Internal Origin Coordinates', 
+        str(INTERNAL_ORIGIN[0]),str(INTERNAL_ORIGIN[1]),str(INTERNAL_ORIGIN[2]),
+        ' ', ' '],
+        ['Project Base Point Coordinates to Internal Origin', 
+        str(convert_values(baspt[0],doc)),
+        str(convert_values(baspt[1],doc)),
+        str(convert_values(baspt[2], doc)), 
+        str(convert_units(basptdistance, document)), baseptangle],
+        ['Survey Point Coordinates to Internal Origin', 
+        str(convert_values(survpt[0], doc)),
+        str(convert_values(survpt[1], doc)),
+        str(convert_values(survpt[2], doc)),
+        str(convert_units(surveydistance, document)), surveyptangle],
+        ['Project Base Point to Survey Delta X', ' ', ' ', ' ',
+        str(convert_units(calculate_delta_x(
+            baspt, survpt), document))],
+        ['Project Base Point to Survey Delta Y', ' ', ' ', ' ',
+        str(convert_units(calculate_delta_y(
+            baspt, survpt), document))],
+        ['Planar Distance between Base Point and Survey Point',
+        ' ', ' ', ' ', 
+        str(convert_units(calculate_horizontal_distance(
+            baspt, survpt), document)),
+            truenorthangle],
+        ['Total Distance between Base Point and Survey Point',
+        ' ', ' ', ' ',
+        str(convert_units(calculate_distance(
+            baspt, survpt), document)),
+            ' '],
+        ['Project Elevation', 
+        ' ', ' ', str(convert_values(baspt[2], doc)),
+        str(convert_units((survpt[2] - baspt[2]), document))]]
+    # Print Table
+    output.print_table(table_data=tbdata, 
+                    title='Project Coordinates and Distances',
+                    columns=
+                        ['Coordinates',
+                        ' X  ',
+                        ' Y  ',
+                        ' Z  ',
+                        ' Distance (' + str(unit_system) + ')  ',
+                        ' ANGLE (°)  '],
+                    formats=['', '' , '', '', '', ''])
+
+def check_model_extents(document=doc):
+    """
+    Checks the extents of all elements in the model.
+    Args:
+        doc (Document): The current document.
+    Returns:
+        None
+    """
+        # _______________________________________Get the bounding box of the 3D view
+    print("")
+    output.print_md('# Checking the extents of the 3D view bounding box')
+    bbox = get_temp_view_bbox(doc)
+    min = (bbox.Min.X, bbox.Min.Y, bbox.Min.Z)
+    max = (bbox.Max.X, bbox.Max.Y, bbox.Max.Z)
+    print("")
+    print(divider)
+    print("")
+    if (calculate_distance(min, INTERNAL_ORIGIN) > EXTENT_DISTANCE or 
+        calculate_distance(max, INTERNAL_ORIGIN) > EXTENT_DISTANCE):
+        output.print_md('{} 3D View Bounding Box extends more than {}'.format(BAD_STRG, CRITERIA_STRG))
+    else:
+        output.print_md('{} 3D View Bounding Box is located less than {}'.format(GOOD_STRG, CRITERIA_STRG))
+        test_score = 1
```

Comment:
Don't use a global variable, just return 0 or 1 and and do the test_score calculations in the main function

Reply:
I like this idea; I am returning an integer of 0 or 1 and add them in the start test function:

`    def startTest(self, doc, output):
        with revit.DryTransaction("Checking Model Extents"):
            check_model_coordinates(doc)
            task_1 = check_model_extents(doc)
            task_2 = check_design_options(doc)
            if task_1 + task_2 < 2:
                output.print_md(
                    ('{} 3D View Bounding Box extends more than {}'.
                     format(BAD_STRG, CRITERIA_STRG)))
                check_linked_elements(doc)
                check_all_elements(doc)
            else:
                output.print_md(
                    ('{} 3D View Bounding Box is located less than {}'.
                     format(GOOD_STRG, CRITERIA_STRG)))`